api/envoy/extensions/filters/network/meta_protocol_proxy/v3/route.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.meta_protocol_proxy.v3;
+
+import "xds/type/matcher/v3/matcher.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.meta_protocol_proxy.v3";
+option java_outer_classname = "RouteProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: Meta Protocol Proxy Route Configuration]
+
+// [#not-implemented-hide:]
+message RouteConfiguration {
+  // The name of the route configuration. For example, it might match route_config_name in
+  // envoy.extensions.filters.network.meta_protocol_proxy.v3.Rds.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The match tree to use when resolving route actions for incoming requests.
+  xds.type.matcher.v3.Matcher route = 2 [(validate.rules).message = {required: true}];
```

Comment:
Nice to see the switch to this (CC @snowp). Have you a working implementation yet based on these matchers? Any reference PR with all the pieces connected together?

In general the strategy of small incremental PRs is good, but it's also good to have some branch or PR to refer to that connects this.

Reply:
Thanks. We have implemented part of it and plan to submit a new PR after the API is merged. All progress and PR will be recorded in #18209.